WireAuthentication/Sources/WireAuthenticationUI/Views/DetermineAuthMethod/DetermineAuthMethodViewModel.swift

```diff
@@ -119,8 +131,12 @@ package final class DetermineAuthMethodViewModel: ObservableObject {
             router.navigate(to: DetermineAuthMethodView.Destination.loginOrRegister(email: email))
 
         case let .loginViaSSO(code):
-            // TODO: [WPB-15946] Genarate URL and handle the error
-            modalDestination = .ssoLogin(url: URL(string: "example")!)
+            do {
+                let url = try await ssoLinkGenerator.generateSSOLink(ssoCode: code)
+                modalDestination = .ssoLogin(url: url)
+            } catch {
+                alert = .invalidSSOLink
+            }
```

Comment:
suggestion: rather than make this whole method async, you could just run this case in a detached Task, but you'd need to ensure that you switch back to main actor to update `modalDestination` and `alert`.

Reply:
done